kv/kvserver: TestLeaseExpirationBasedDrainTransferWithExtension failed

kv/kvserver.TestLeaseExpirationBasedDrainTransferWithExtension [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11789413?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11789413?buildTab=artifacts#/) on release-23.1.11-rc @ [b2012193c9558ed2f6541da27469a8c5ecca52ae](https://github.com/cockroachdb/cockroach/commits/b2012193c9558ed2f6541da27469a8c5ecca52ae):


```
=== RUN   TestLeaseExpirationBasedDrainTransferWithExtension
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestLeaseExpirationBasedDrainTransferWithExtension2127012733
    test_log_scope.go:79: use -show-logs to present logs inline
    client_replica_test.go:1806: condition failed to evaluate within 3m45s: [NotLeaseHolderError] lease held by different store; r64: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):2 seq=2 start=1694777961.282534834,0 exp=1694778193.624914971,0 pro=1694778187.624914971,0
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestLeaseExpirationBasedDrainTransferWithExtension2127012733
--- FAIL: TestLeaseExpirationBasedDrainTransferWithExtension (233.09s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBasedDrainTransferWithExtension.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31574

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -31,6 +31,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeedcache"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/allocator/storepool"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
@@ -1439,23 +1440,36 @@ func setupLeaseTransferTest(t *testing.T) *leaseTransferTest {
 		}
 	}
 
-	l.tc = testcluster.StartTestCluster(t, 2,
-		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationManual,
-			ServerArgs: base.TestServerArgs{
-				Knobs: base.TestingKnobs{
-					Store: &kvserver.StoreTestingKnobs{
-						EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
-							TestingEvalFilter: testingEvalFilter,
-						},
-						TestingProposalFilter:                testingProposalFilter,
-						LeaseTransferBlockedOnExtensionEvent: leaseTransferBlockedOnExtensionEvent,
+	const numNodes = 2
+	serverArgs := make(map[int]base.TestServerArgs, numNodes)
+	for i := 0; i < numNodes; i++ {
+		st := cluster.MakeClusterSettings()
+		// leaseTransferTest tests manually control the clock and may induce clock
+		// jumps. Disable the suspect timer to prevent nodes from becoming suspect
+		// and being excluded as lease transfer targets when we bump clocks.
+		storepool.TimeAfterStoreSuspect.Override(context.Background(), &st.SV, 0)
+
+		serverArgs[i] = base.TestServerArgs{
+			Settings: st,
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
+						TestingEvalFilter: testingEvalFilter,
 					},
-					Server: &server.TestingKnobs{
-						WallClock: l.manualClock,
-					},
+					TestingProposalFilter:                testingProposalFilter,
+					LeaseTransferBlockedOnExtensionEvent: leaseTransferBlockedOnExtensionEvent,
+				},
+				Server: &server.TestingKnobs{
+					WallClock: l.manualClock,
 				},
 			},
+		}
+	}
+
+	l.tc = testcluster.StartTestCluster(t, numNodes,
+		base.TestClusterArgs{
+			ReplicationMode:   base.ReplicationManual,
+			ServerArgsPerNode: serverArgs,
 		})
 	key := l.tc.ScratchRangeWithExpirationLease(t)
 	l.tc.AddVotersOrFatal(t, key, l.tc.Target(1))

